src/els_compiler_diagnostics.erl

```diff
@@ -92,11 +92,27 @@ parse(Uri) ->
   {ok, Epp} = epp:open([ {name, FileName}
                        , {includes, els_config:get(include_paths)}
                        ]),
-  Res = [diagnostic(range(Line), Module, Desc, ?DIAGNOSTIC_ERROR)
+  Res = [epp_diagnostic(Line, Module, Desc)
          || {error, {Line, Module, Desc}} <- epp:parse_file(Epp)],
   epp:close(Epp),
   Res.
 
+%% Possible cases to handle
+%% ,{error,{19,erl_parse,["syntax error before: ","'-'"]}}
+%% ,{error,{1,epp,{error,1,{undefined,'MODULE',none}}}}
+%% ,{error,{3,epp,{error,"including nonexistent_macro.hrl is not allowed"}}}
+%% ,{error,{3,epp,{include,file,"yaws.hrl"}}}
+-spec epp_diagnostic(integer(), module(), any()) ->
+        els_diagnostics:diagnostic().
+epp_diagnostic(Line, epp, {error, Desc}) ->
+  diagnostic(range(Line), epp, Desc, ?DIAGNOSTIC_ERROR);
+epp_diagnostic(Line, epp, {error, Line, Desc}) ->
+  diagnostic(range(Line), epp, Desc, ?DIAGNOSTIC_ERROR);
+epp_diagnostic(Line, epp, Desc) ->
+  diagnostic(range(Line), epp, Desc, ?DIAGNOSTIC_ERROR);
+epp_diagnostic(Line, erl_parse, Desc) ->
+  diagnostic(range(Line), erl_parse, Desc, ?DIAGNOSTIC_ERROR).
```

Comment:
I think we only need to handle the `{error, Line, Desc}` case specifically. It seems to be a bug in `epp:format_error/1` that it crashes for that. The other cases should work just fine [looking at the implementation](https://github.com/erlang/otp/blob/4f580ad1bbc18d798dd3391bab7c714902f9b996/lib/stdlib/src/epp.erl#L189-L233). Especially `{error, Desc}` that is formatted by epp differently than it would be in here.

Reply:
Reported upstream https://bugs.erlang.org/browse/ERL-1310

Reply:
The comments in line 101-104 are actual cases seen, from the test.  So the other cases *are* needed, I think.

Reply:
The other cases should work just fine with what `diagnostic` already does - i.e. calling `Mod:format_error(Reason)`. It's just that this one seems to crash on that call.